app/Leaf/VAMCActiveDirectory.php

```diff
@@ -73,52 +73,61 @@ public function importADData(string $file): string
 
         $count = 0;
 
-        foreach ($write_data as $employee) {
-            if (
+	foreach ($write_data as $employee) {
+		
+		if (
                 isset($employee['lname'])
                 && $employee['lname'] != ''
                 && isset($employee['loginName'])
                 && $employee['loginName'] != ''
-            ) {
+	    ) {
+		    if(empty($employee['midIni'])){
+			    $employee['midIni'] = '';
+		    }
+
+            // there is incomplete data in ad in some cases.
+		    if(empty($employee['fname'])){
+			    $employee['fname'] = '';
+		    }
```

Comment:
Is there a reason to do this here and not in line. Ie $this->users[$id]['fname'] = !empty($employee['fname'] ? $employee['fname'] : ''; Same for the midIni. We are doing that for email, phone, pager, etc.